src/db/oracledb/jobs-dao.js

```diff
@@ -56,4 +120,257 @@ const getJobByJobId = async (internalId, jobId) => {
+ * @returns {string} Query result, null if success
+ */
+const graduateJob = async (connection, osuId, body, operation) => {
+  const binds = standardBinds(osuId, body, graduateBinds);
+  const queryName = operation === 'update' ? 'graduateUpdateJob' : 'graduateCreateJob';
+
+  const { outBinds: { result } } = await connection.execute(
+    contrib[queryName](binds),
+    binds,
+  );
+  return result;
+};
+
+/**
+ * Determines if changeReasonCode is valid
+ *
+ * @param {object} connection oracledb connection
+ * @param {string} changeReasonCode Change reason code from request body
+ * @returns {boolean} True if changeReasonCode is valid
+ */
+const isValidChangeReasonCode = async (connection, changeReasonCode) => {
+  const { rows } = await connection.execute(
+    contrib.validateChangeReasonCode(),
+    { changeReasonCode },
+  );
+  return rows[0].count > 0;
+};
+
+/**
+ * Determines which Epaf to execute based on fields in body
+ *
+ * @param {string} operation 'create' to create a new record, 'update' to update an existing one
+ * @param {string} osuId OSU ID of a person
+ * @param {object} body Request body
+ * @param {string} internalId Internal ID of a person
+ * @returns {Error} returns error or null if no error occurred
+ */
+const createOrUpdateJob = async (operation, osuId, body) => {
+  const connection = await getConnection('banner');
+  try {
+    let error;
+    const { studentEmployeeInd, changeReason: { code: changeReasonCode } } = body;
+    const employmentType = studentEmployeeInd ? 'student' : 'graduate';
+
+    if (employmentType === 'student' && !/^C5[0-2]/.test(body.positionNumber)) {
+      return new Error('Position number for students must start with C50, C51, or C52');
+    }
+
+    if (_.includes(['TERME', 'TERMJ'], changeReasonCode)) {
+      error = await terminateJob(connection, osuId, body);
+    } else {
+      if (!await isValidChangeReasonCode(connection, changeReasonCode)) {
+        return new Error(`Invalid change reason code ${changeReasonCode}`);
+      }
+
+      if (operation === 'update') {
+        if (changeReasonCode === 'NONE') {
+          error = await updateLaborChangeJob(connection, osuId, body);
+        } else if (changeReasonCode === 'BREAP') {
+          if (employmentType === 'student') {
+            error = await studentJob(connection, osuId, body, operation);
+          } else if (employmentType === 'graduate') {
+            error = await graduateJob(connection, osuId, body, operation);
+          }
+        } else {
+          error = await updateJob(connection, osuId, body);
+        }
+      } else if (employmentType === 'student') {
+        error = await studentJob(connection, osuId, body, operation);
+      } else if (employmentType === 'graduate') {
+        error = await graduateJob(connection, osuId, body, operation);
+      }
+    }
+
+    if (!error) {
+      connection.commit();
+      return error;
+    }
+    throw new Error(error);
```

Comment:
I guess it should be `return` instead of `throw`?

Reply:
`return` or `throw` is the difference of returning a `400` or `500` error. I think if we get a db error we should return 500 because the error is likely on our end and not from the users input. Thoughts?

Reply:
Make sense.